cpp/ql/src/experimental/Security/CWE/CVE-094/EmscriptenRunScriptTaint.ql

```diff
@@ -0,0 +1,54 @@
+/**
+ * @name Uncontrolled data used in Emscripten run script commands
+ * @description Using user-supplied data in Emscripten run script
+ *              commands, without proper sanitization, can make code
+ *              vulnerable to JavaScript code injection.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id cpp/emscripten/run-script-injection
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-094
+ */
+
+import cpp
+import semmle.code.cpp.dataflow.new.TaintTracking
+import DataFlow::PathGraph
+
+class EmscriptenRunScriptTaintConfig extends TaintTracking::Configuration {
+  EmscriptenRunScriptTaintConfig() { this = "EmscriptenRunScriptTaintConfig" }
+
+  override predicate isSource(DataFlow::Node source) {
+    exists(Parameter p | source.asParameter() = p)
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(FunctionCall fc |
+      sink.asExpr() = fc.getArgument(0) and
+      (
+        fc.getTarget().hasGlobalName("emscripten_run_script") or
+        fc.getTarget().hasGlobalName("emscripten_run_script_int") or
+        fc.getTarget().hasGlobalName("emscripten_run_script_string") or
+        fc.getTarget().hasGlobalName("emscripten_async_run_script") or
+        fc.getTarget().hasGlobalName("emscripten_async_load_script")
```

Comment:
Why is `emscripten_async_load_script` included here? Looking at the documentation, it seems that this only loads a script, but doesn't do anything with it. Wouldn't this only cause problems if the script is actually executed? If that's the case, it seems more appropriate to have calls to `emscripten_async_load_script` as an additional taint step.